include/mango_v3.hpp

```diff
@@ -140,6 +142,39 @@ struct MangoAccount {
   uint8_t padding[5];
 };
 
+class MangoAccount {
+  // Creating a MangoAccount might involve non-trivial work like calling RPC
+  // methods, so we provide explict `MangoAccount::from()` overloaded set of
+  // functions and make the public constructor un-callable.
+  struct NotPubliclyConstructible {};
+
+ public:
+  MangoAccount(NotPubliclyConstructible /* npc */,
+               const MangoAccountInfo &accountInfo) noexcept
+      : accountInfo_(accountInfo) {}
+  // Use a prefetched accountInfo
+  static std::unique_ptr<MangoAccount> from(
+      const MangoAccountInfo &accountInfo) noexcept {
+    return std::make_unique<MangoAccount>(NotPubliclyConstructible{},
+                                          accountInfo);
+  }
+  // Fetch `accountInfo` from `endpoint` and decode it
+  static std::unique_ptr<MangoAccount> from(
+      const solana::PublicKey &pubKey,
+      const std::string &endpoint = MAINNET.endpoint) {
+    auto connection = solana::rpc::Connection(endpoint);
+    const auto &accountInfo =
+        connection.getAccountInfo<MangoAccountInfo>(pubKey.toBase58());
+    return std::make_unique<MangoAccount>(NotPubliclyConstructible{},
+                                          accountInfo);
+  }
+  i80f48 getLiquidationPrice() { return {}; }
+  // TODO: Add methods to calculate health ratio
+ private:
+  const MangoAccountInfo &accountInfo_;
```

Comment:
don't think it's good that this one is const and private. i know that a lot of people claim that you should always keep everything and private and so on, but i really hate it, when it limits the usability of the code. 

it's really handy to be able to create a copy and manipulate some values in there to simulate operations like withdrawal margin checks client side. we do it frequently in mango ui.

Reply:
Will update this to remove this `const` for now. I will add a public method to retrieve  the `AccountInfo` of a given `MangoAccount`

Reply:
try to use as little encapsulation as necessary and not default to the maximum. it just creates code that no one wants to read.